docs/src/extend/custom-rules.md

```diff
@@ -635,73 +679,45 @@ module.exports = {
 };
 ```
 
-In the preceding example, the error level is assumed to be the first argument. It is followed by the first optional argument, a string which may be either `"always"` or `"never"`. The final optional argument is an object, which may have a Boolean property named `exceptRange`.
+In the preceding example, the error level is assumed to be the first argument. It is followed by the first optional argument, a string which may be either `"always"` or `"never"`. The final optional argument is an object, which may have a boolean property named `exceptRange`.
 
 To learn more about JSON Schema, we recommend looking at some examples in [website](https://json-schema.org/learn/) to start, and also reading [Understanding JSON Schema](https://json-schema.org/understanding-json-schema/) (a free ebook).
 
-**Note:** Currently you need to use full JSON Schema object rather than array in case your schema has references ($ref), because in case of array format ESLint transforms this array into a single schema without updating references that makes them incorrect (they are ignored).
```

Comment:
There are two ways to define a schema. Quoting the same page:

https://github.com/eslint/eslint/blob/f9f195ef12deb114fb86763010a23ea0cb4c78d1/docs/src/extend/custom-rules.md?plain=1#L612-L614

So this note is saying that if your schema contains references (identified by the property key `$ref`), then you have to use a full JSON schema rather than the simplified array format supported by ESLint.